### DISCUSSION TITLE
Valhalla/Shortest distance route

### DISCUSSION THREAD

**Discussion by ArterialMapping** [mthh/routingpy#62] [Q&A]
I am attempting to compute shortest travel time-based routes as well as shortest distance-based routes. Within the [Valhalla API referenced](https://github.com/valhalla/valhalla/blob/master/docs/api/turn-by-turn/api-reference.md#costing-options) as linked on routingpy [docs](https://routingpy.readthedocs.io/en/latest/#valhalla) "shortest" is listed as a costing option in which it "Changes the metric to quasi-shortest, i.e. purely distance-based costing. 

However, when running this option I am not achieving the purely shortest distance-based route. For instance, see an example from the US State of Virginia below. There are alternatives further to the west that should be using the selected area to route through; however, they are not. 

![image](https://user-images.githubusercontent.com/85142174/159599962-37c500b6-fc23-48a9-982c-21e676673923.png)

So I am wondering if I am setting this up in my code incorrectly. Code is as described below for accessing the docker client and detailing the client. directions.

For using option:

        coords = [[X, Y], [-77.435572, 37.540440]]

        client = Valhalla(base_url='http://localhost:8002')

        route = client.directions(locations=coords, profile='auto', option='shortest')

I have also changed my preference to equal "shortest" instead of "fastest" which is what I had it set to for travel time. Alike to the description/image this also did not achieve the desired result. Here would be the code for preference: 

        coords = [[X, Y], [-77.435572, 37.540440]]

        client = Valhalla(base_url='http://localhost:8002')

        route = client.directions(locations=coords, profile='auto', preference='shortest')

So my question is, what would be the best method to achieve the actual shortest distance if the options above did not work?

**Comment by nilsnolde** [comment]
this is correct: `route = client.directions(locations=coords, profile='auto', preference='shortest')`

I can't make out a proof that it doesn't work. for that pls show 2 routes, one with `shortest`, one with without. did you update routingpy? I released `0.4.0` on pypi a few days ago.

it's also not really shortest, just shortest-ish. on longer routes it won't be really shortest, since there's shortcuts involved that we don't care about in valhalla.

**Reply by ArterialMapping** [reply]
The example below is what preference='fastest' looks like in a similar route location (although this image is a bit more detailed). It looks as if they generally outputted the same routing. 

![image](https://user-images.githubusercontent.com/85142174/159800220-b3261264-9624-4e0f-b35c-1a5565f1d32e.png)

And this is with no preference set: 

![image](https://user-images.githubusercontent.com/85142174/159800166-84377b41-dd30-4f21-a88e-22ef8dff06c3.png)

Am I unable to use options='shortest' as indicated in the routingpy and Valhalla documentation? As attributed in the Valhalla documentation this seems to be the correct way to achieve the true shortest distance routes.  

And I do not currently have the update as I am using conda.

Edit: Have now installed under a new interpreter the update to 0.4.0. Still no luck with retrieving the shortest distance routes.

**Reply by ArterialMapping** [reply]
Switched to ORS, works perfectly now!
